### ISSUE TITLE
Tripleta loss finetuning?

### ISSUE THREAD

**Issue by Davidc2525** [timesler/facenet-pytorch#230]
Can you leave an example of how to do finetuning with triplet loss?

**Comment by ihsingh2** [comment]
Hey,

Given a list of triplets, you need to define a dataset for them. You can refer to [these](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html#dataset-class) [two](https://github.com/andreasveit/triplet-network-pytorch/blob/master/triplet_image_loader.py#L11) examples to get an idea of how to define custom datasets. Once that is done, you can setup a dataloader for this dataset, and use it in your training loop. You can use [torch.nn.TripletMarginLoss](https://pytorch.org/docs/stable/generated/torch.nn.TripletMarginLoss.html) for the loss function.

However, if you are using training.pass_epoch, do note that it expects to unpack a batch of single images at a time (see [this](https://github.com/timesler/facenet-pytorch/blob/8ee5326dc904f88846ca014a3d047b08392bd12f/models/utils/training.py#L95)). You will have to modify it to unpack a batch of tuple of three images and evaluate each image on the model individually, before passing on to the loss function.

Hope that helps!

**Comment by fn-hide** [comment]
Any updates?
